src/provider/controller/ContextHandler.ts

```diff
@@ -40,8 +40,8 @@ export class ContextHandler {
      * @param context Context to send
      * @param source Window sending the context. It won't receive the broadcast
      */
-    public broadcast(context: Context, source: AppWindow): void {
-        this.broadcastOnChannel(context, source, source.channel);
+    public async broadcast(context: Context, source: AppWindow): Promise<void> {
+        return this.broadcastOnChannel(context, source, source.channel);
```

Comment:
I could probably check myself, but where does this await chain end? Is it once the broadcast reaches the provider, when the provider has sent to all clients, when the clients have all completed their callback, or some other timing? the spirit of broadcast is that it's fire-and-forget, so want to be sure we're only awaiting what is strictly necessary to catch problems.

Reply:
Fine as discussed